### PR TITLE
[SYCL] [DOC] Add note on DISABLE_SYCL_INSTRUMENTATION_METADATA macro

### PR DESCRIPTION
**Pull request by s-kanaev**
Signed-off-by: Sergey Kanaev <sergey.kanaev@intel.com>

### CODE REVIEW

File: sycl/doc/PreprocessorMacros.md [hunk 1]
```diff
@@ -19,6 +19,13 @@ This behavior is not following the SYCL spec since `constant_ptr` conversions to
 the underlying pointer types return pointers without any additional qualifiers
 so it's disabled by default.
 
+
+### `DISABLE_SYCL_INSTRUMENTATION_METADATA`
```
Comment (bader) [inline sycl/doc/PreprocessorMacros.md:23]:
```suggestion
### `DISABLE_SYCL_INSTRUMENTATION_METADATA`
```
Please, keep one-line separator.

File: sycl/doc/PreprocessorMacros.md [hunk 2]
```diff
@@ -19,6 +19,13 @@ This behavior is not following the SYCL spec since `constant_ptr` conversions to
...
+
+This macro is used to disable passing of code location metadata to `submit*`
+methods of `cl::sycl::queue` class.
+
+
```
Comment (bader) [inline sycl/doc/PreprocessorMacros.md:28]:
```suggestion

```

File: sycl/doc/PreprocessorMacros.md [hunk 3]
```diff
@@ -19,6 +19,13 @@ This behavior is not following the SYCL spec since `constant_ptr` conversions to
 the underlying pointer types return pointers without any additional qualifiers
 so it's disabled by default.
 
+
+### `DISABLE_SYCL_INSTRUMENTATION_METADATA`
+
+This macro is used to disable passing of code location metadata to `submit*`
+methods of `cl::sycl::queue` class.
```
Comment (bader) [inline sycl/doc/PreprocessorMacros.md:26]: I think the intent of this macro is much wider than `cl::sycl::queue` class. Probably we should generalize the wording.

File: sycl/doc/PreprocessorMacros.md (hunk 1 above)
Comment (s-kanaev) [reply sycl/doc/PreprocessorMacros.md]: Done

File: sycl/doc/PreprocessorMacros.md (hunk 3 above)
Comment (s-kanaev) [reply sycl/doc/PreprocessorMacros.md]: Done.

File: sycl/doc/PreprocessorMacros.md (hunk 2 above)
Comment (s-kanaev) [reply sycl/doc/PreprocessorMacros.md]: Done

**Comment by s-kanaev** [comment]
@tovinkere , could you, please, review?

**Review by tovinkere — DISMISSED** [review]
LGTM

**Comment by s-kanaev** [comment]
Resolved merge conflict

**Pull request merged**